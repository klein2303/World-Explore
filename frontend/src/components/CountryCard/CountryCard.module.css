.cardcontainer {
    background-color: #e4e1f7;
    height: 250px;
    width: 250px;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: space-between;
    border-radius: 10px;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.1);
}

/* Dark mode Card Container */
body[data-theme='dark'] .cardcontainer {
    background-color: #8A86A4; 
    color: white;
    box-shadow: 0 4px 6px rgba(0, 0, 0, 0.5); 
}

.cardcontainer:hover{
    background-color: #ddd9e3;
}

/* Dark mode hover effect */
body[data-theme='dark'] .cardcontainer:hover {
    background-color: #827f7f; 
}

.cardimage {
    height: 200px;
    width: 250px;
    border-radius: 10px 10px 0px 0px;
    overflow: hidden;
}

.information {
    width: 220px;
    height: 50px;
    display: flex;
    flex-direction: row;
    align-items: center;
    justify-content: space-between;
    color: #424242;
}

/* Dark mode information section */
body[data-theme='dark'] .information {
    color: white; 
}

.icon {
    cursor: pointer;
    font-size: 22px;
}

.icon:hover {
    fill: #424242;
    font-size: 25px;
}

/* Dark mode icon hover effect */
body[data-theme='dark'] .icon:hover {
    fill: white; 
}

.name {
    font-size: 15px;
}

@media (max-width: 810px) {
    .cardcontainer {
        margin: 12px;
    }
}
