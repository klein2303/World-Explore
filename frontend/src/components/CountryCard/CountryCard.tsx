import {useEffect, useState} from "react";
import styles from "./CountryCard.module.css";
import {LuMapPin} from "react-icons/lu";
import {Link} from "react-router-dom";

interface CountryCardProps {
    name: string;
    image: string;
}

const CountryCard = ({name, image}: CountryCardProps) => {
    const [isFilled, setIsFilled] = useState<boolean>(false);

    const fillMapPin = () => {
        setIsFilled(!isFilled);

        if (isFilled) {
            localStorage.removeItem(name);
        } else {
            localStorage.setItem(name, "filled");
        }
    };

    useEffect(() => {
        const filled = localStorage.getItem(name);
        if (filled) {
            setIsFilled(true);
        }
    }, [name]);

    return (
        <main className={styles.cardcontainer} role="region" aria-labelledby={`country-name-${name}`}>
            <Link to={`/${name.toLowerCase()}`}>
                <img src={image} alt={`image of the country ${name}`} className={styles.cardimage} aria-hidden="true" />
            </Link>
            <div className={styles.information}>
                <p id={`country-name-${name}`} className={styles.name}>
                    {name}
                </p>
                {/* Conditionally render the map pin icon based on whether it's filled */}
                {isFilled ? (
                    <LuMapPin
                        onClick={fillMapPin}
                        className={styles.filled}
                        role="button"
                        aria-pressed={isFilled}
                        aria-label={`Unmark ${name} from visited`}
                    />
                ) : (
                    <LuMapPin
                        onClick={fillMapPin}
                        className={styles.icon}
                        role="button"
                        aria-pressed={isFilled}
                        aria-label={`Mark ${name} as visited`}
                    />
                )}
            </div>
        </main>
    );
};

export default CountryCard;
