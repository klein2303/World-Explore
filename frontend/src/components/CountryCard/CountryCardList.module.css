.cardlist {
    height: fit-content;
    width: 900px;

    display: flex;
    flex-direction: row;
    gap: 16px;
    flex-wrap: wrap;

    justify-content: center;
    align-content: flex-start;
}

.listitem {
    width: fit-content;
    height: fit-content;
}

@media (max-width: 1200px) {
    .cardlist {
        width: 700px;
        height: 600px;
        gap: 8px;
    }
}

@media (max-width: 900px) {
    .cardlist {
        width: 500px;
        height: 600px;
        gap: 8px;
    }
}

@media (max-width: 620px) {
    .cardlist {
        width: 300px;
        height: 600px;
        gap: 30px;
    }
}
