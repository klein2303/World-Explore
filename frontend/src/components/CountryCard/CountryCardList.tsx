import styles from "./CountryCardList.module.css";
import CountryCard from "./CountryCard";
import {CountryType} from "../../types/CountryTypes";

interface CountryCardListProps {
    countries: CountryType[];
}

const CountryCardList = ({countries}: CountryCardListProps) => {
    return (
        <main className={styles.cardlist} role="list">
            {countries.map((country: CountryType) => (
                <div role="listitem" key={country.name} className={styles.listitem}>
                    <CountryCard name={country.name} image={country.image} />
                </div>
            ))}
        </main>
    );
};

export default CountryCardList;
