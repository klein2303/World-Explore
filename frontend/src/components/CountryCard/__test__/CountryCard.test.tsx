import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import { describe, it, expect } from "vitest";
import CountryCard from "../CountryCard";
import { MemoryRouter } from "react-router-dom";

describe("CountryCard Tests", () => {
    it("renders the country card correctly", () => {
        // Render the component
        const { asFragment } = render(
            <MemoryRouter>
                <CountryCard name="Australia" image="https://example.com/australia.jpg" />,
            </MemoryRouter>,
        );

        // Take a snapshot of the initial render
        expect(asFragment()).toMatchSnapshot();

        // Check if the country name is in the document
        expect(screen.getByText("Australia")).toBeInTheDocument();

        // Check if the image is in the document
        const countryImage = screen.getByAltText("image of the country Australia");
        expect(countryImage).toBeInTheDocument();
        expect(countryImage).toHaveAttribute("src", "https://example.com/australia.jpg");

        // Check if the map pin icon is in the document
        const mapPinIcon = screen.getByLabelText("Mark Australia as visited");
        expect(mapPinIcon).toBeInTheDocument();
    });

    it("toggles the map pin icon when clicked", () => {
        render(
            <MemoryRouter>
                <CountryCard name="Australia" image="https://example.com/australia.jpg" />,
            </MemoryRouter>,
        );

        // Check if the map pin icon is in the document
        const mapPinIcon = screen.getByLabelText("Mark Australia as visited");
        expect(mapPinIcon).toBeInTheDocument();

        // Fire event to mark the map pin icon
        fireEvent.click(mapPinIcon);

        // Check if the map pin icon is toggled
        const filledMapPinIcon = screen.getByLabelText("Unmark Australia from visited");
        expect(filledMapPinIcon).toBeInTheDocument();

        // Fire event to unmark the map pin icon
        fireEvent.click(filledMapPinIcon);

        // Check if the map pin icon is toggled back
        const unfilledMapPinIcon = screen.getByLabelText("Mark Australia as visited");
        expect(unfilledMapPinIcon).toBeInTheDocument();
    });
});
