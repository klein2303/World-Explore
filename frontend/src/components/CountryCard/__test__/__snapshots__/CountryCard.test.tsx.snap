// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`CountryCard > opens the modal when the pencil icon is clicked 1`] = `
<DocumentFragment>
  <main
    aria-label="country-name-France"
    class="_cardcontainer_043c24"
    role="region"
  >
    <a
      href="/Countries/France"
    >
      <img
        alt="image of the country France"
        aria-hidden="true"
        class="_cardimage_043c24"
        src="https://via.placeholder.com/150"
      />
    </a>
    <div
      aria-description="Information and actions for France"
      class="_information_043c24"
      role="group"
    >
      <p
        class="_name_043c24"
        id="country-name-France"
      >
        France
      </p>
      <svg
        aria-description="Write review for France"
        class="_icon_043c24"
        fill="currentColor"
        height="1em"
        role="button"
        stroke="currentColor"
        stroke-width="0"
        tabindex="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M964.256 49.664C929.392 16.256 890.933-.672 849.877-.672c-64.192 0-111.024 41.472-123.841 54.176-18.032 17.856-633.152 633.2-633.152 633.2a33.011 33.011 0 0 0-8.447 14.592C70.565 752.559 1.077 980.016.387 982.304c-3.567 11.648-.384 24.337 8.208 32.928a32.336 32.336 0 0 0 22.831 9.44c3.312 0 6.655-.496 9.919-1.569 2.352-.767 237.136-76.655 275.775-88.19a32.736 32.736 0 0 0 13.536-8.033c24.416-24.128 598.128-591.456 636.208-630.784 39.392-40.592 58.96-82.864 58.208-125.616-.784-42.208-21.248-82.848-60.816-120.816zM715.845 155.84c16.304 3.952 54.753 16.862 94.017 56.479 39.68 40.032 50.416 85.792 52.416 96.208-125.824 125.168-415.456 411.728-529.632 524.672-10.544-24.56-27.584-54.144-54.993-81.76-33.471-33.728-67.536-52.783-93.808-63.503 112.992-113.008 408.08-408.224 532-532.096zM140.39 741.95c17.584 4.672 54.111 18.224 91.344 55.76 28.672 28.912 42.208 60.8 48.288 80.24-44.48 14.304-141.872 47.92-203.76 67.872 18.336-60.336 49.311-154.304 64.128-203.872zm780.031-491.584a1748.764 1748.764 0 0 1-6.065 6.16c-10.113-26.049-27.857-59.52-58.577-90.496-31.391-31.648-63.231-50.32-88.75-61.36 2.175-2.16 3.855-3.857 4.511-4.496 3.664-3.617 36.897-35.376 78.32-35.376 23.84 0 47.248 10.88 69.617 32.32 26.511 25.424 40.175 50.512 40.624 74.592.431 24.576-12.913 51.04-39.68 78.656z"
        />
      </svg>
      <div
        data-testid="journal-entry-modal"
      >
        Journal Entry Modal
      </div>
    </div>
  </main>
</DocumentFragment>
`;

exports[`CountryCard > renders correctly with the given props 1`] = `
<DocumentFragment>
  <main
    aria-label="country-name-France"
    class="_cardcontainer_043c24"
    role="region"
  >
    <a
      href="/Countries/France"
    >
      <img
        alt="image of the country France"
        aria-hidden="true"
        class="_cardimage_043c24"
        src="https://via.placeholder.com/150"
      />
    </a>
    <div
      aria-description="Information and actions for France"
      class="_information_043c24"
      role="group"
    >
      <p
        class="_name_043c24"
        id="country-name-France"
      >
        France
      </p>
      <svg
        aria-description="Write review for France"
        class="_icon_043c24"
        fill="currentColor"
        height="1em"
        role="button"
        stroke="currentColor"
        stroke-width="0"
        tabindex="0"
        viewBox="0 0 1024 1024"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M964.256 49.664C929.392 16.256 890.933-.672 849.877-.672c-64.192 0-111.024 41.472-123.841 54.176-18.032 17.856-633.152 633.2-633.152 633.2a33.011 33.011 0 0 0-8.447 14.592C70.565 752.559 1.077 980.016.387 982.304c-3.567 11.648-.384 24.337 8.208 32.928a32.336 32.336 0 0 0 22.831 9.44c3.312 0 6.655-.496 9.919-1.569 2.352-.767 237.136-76.655 275.775-88.19a32.736 32.736 0 0 0 13.536-8.033c24.416-24.128 598.128-591.456 636.208-630.784 39.392-40.592 58.96-82.864 58.208-125.616-.784-42.208-21.248-82.848-60.816-120.816zM715.845 155.84c16.304 3.952 54.753 16.862 94.017 56.479 39.68 40.032 50.416 85.792 52.416 96.208-125.824 125.168-415.456 411.728-529.632 524.672-10.544-24.56-27.584-54.144-54.993-81.76-33.471-33.728-67.536-52.783-93.808-63.503 112.992-113.008 408.08-408.224 532-532.096zM140.39 741.95c17.584 4.672 54.111 18.224 91.344 55.76 28.672 28.912 42.208 60.8 48.288 80.24-44.48 14.304-141.872 47.92-203.76 67.872 18.336-60.336 49.311-154.304 64.128-203.872zm780.031-491.584a1748.764 1748.764 0 0 1-6.065 6.16c-10.113-26.049-27.857-59.52-58.577-90.496-31.391-31.648-63.231-50.32-88.75-61.36 2.175-2.16 3.855-3.857 4.511-4.496 3.664-3.617 36.897-35.376 78.32-35.376 23.84 0 47.248 10.88 69.617 32.32 26.511 25.424 40.175 50.512 40.624 74.592.431 24.576-12.913 51.04-39.68 78.656z"
        />
      </svg>
    </div>
  </main>
</DocumentFragment>
`;
